include/rocksdb/file_system.h

```diff
@@ -1097,6 +1097,15 @@ class FSDirectory {
   // Fsync directory. Can be called concurrently from multiple threads.
   virtual IOStatus Fsync(const IOOptions& options, IODebugContext* dbg) = 0;
 
+  // Fsync after renaming a file. Depends on the filesystem, it may fsync
+  // directory or just the renaming file (e.g. btrfs). By default, it just calls
+  // directory fsync.
+  virtual IOStatus FsyncForRename(const std::string& filename,
+                                  const IOOptions& options,
+                                  IODebugContext* dbg) {
+    return Fsync(options, dbg);
+  }
+
```

Comment:
Because this is only for a file rename. For new file creation (or deletion) we still just call Fsync(). We want to be explicit for that. But I agree that putting that information in IOOption would be more flexible.